lib/hiera/backend/vault_backend.rb

```diff
@@ -61,13 +82,21 @@ def lookup_generic(key, scope)
           return nil if secret.nil?
 
           Hiera.debug("[hiera-vault] Read secret: #{key}")
-          if @config[:default_field]
+          if @config[:default_field] and secret.data.has_key?(@config[:default_field].to_sym) and secret.data.length == 1
```

Comment:
Ignoring data that is in vault does not make sense to me. Ignoring the additional fields makes it impossible to retrieve the contents of those fields with hiera when using a default_field. I'd leave it to the consumer to decide whether to ignore the additional fields.
But I'll make the non-ignoring behavior optional as well, to make sure default behavior stays backwards compatible.